### DISCUSSION TITLE
afterStateUpdated() function does not work when hidden() is true

### DISCUSSION THREAD

**Discussion by fatihdurmaz** [filamentphp/filament#12494] [General]
Hello, I want to empty the kisim_sayisi field when the kismi_teklife_acik_mi field is closed. If it is visible, it works, but if it is not visible, it does not assign null. In short, afterStateUpdated() does not work when hidden() is true.

                                Toggle::make('kismi_teklife_acik_mi')
                                    ->live()
                                    ->requiredWith('kisim_sayisi')
                                    ->inline(false)
                                    ->afterStateUpdated(function (Get $get, Set $set) {
                                        if (($get('kismi_teklife_acik_mi') == false)) {
                                            $set('kisim_sayisi', null);
                                        }
                                    })
                                    ->label('İhale Kısmi Teklife Açık Mı?'),

                                TextInput::make('kisim_sayisi')
                                    ->label('Kısım Sayısı')
                                    ->requiredWith('kismi_teklife_acik_mi')
                                    ->numeric()
                                    ->validationAttribute('Kısım Sayısı')
                                    ->hidden(
                                        fn (Get $get): bool => !$get('kismi_teklife_acik_mi')
                                    ),

**Comment by nicolawebdev** [comment]
+1, same issue here

**Comment by jcleggfai** [comment]
+1, just found this issue to my detriment. For anyone facing this issue adding 
`->dehydratedWhenHidden()` to the select does fix the issue. This is a nasty bug to run into for something as simple as assigning null to a select field with hidden fields on the same form. Is anyone from Filament working to fix this? Is there a proper resolution to the issue?
